modules/javafx.controls/src/test/java/test/javafx/scene/control/CellEditEventOfTreeTableColumnTest.java

```diff
@@ -0,0 +1,199 @@
+        assertNull("rowValue must be null for null pos", ev.getRowValue());
+    }
+
+// ------------- event source
+
+    @Ignore("JDK-8271474")
+    @Test
+    public void testNullTable() {
+        new CellEditEvent<Object, Object>(null, // null table must not throw NPE
+                new TreeTablePosition<>(null, -1, null), editAnyEvent(), null);
+    }
+
+    @Test
+    public void testCellEditEventDifferentSource() {
+        assertCellEditEvent(new TreeTableView<>());
+    }
+
+    @Test
+    public void testCellEditEventSameSource() {
+        assertCellEditEvent(table);
+    }
+
+    @Ignore("JDK-8271474")
+    @Test
+    public void testCellEditEventNullSource() {
+        assertCellEditEvent(null);
+    }
+
+    /**
+     * Creates a CellEditEvent with the given source, not-null position and asserts
+     * all properties of the event.
+     *
+     * @param source the source of the event
+     */
+    private void assertCellEditEvent(TreeTableView<String> source) {
+        int editingRow = 1;
+        String editedValue = "edited";
+        TreeItem<String> rowValue = table.getTreeItem(editingRow);
+        String oldValue = rowValue.getValue();
+        TreeTablePosition<String, String> pos = new TreeTablePosition<>(table, editingRow, editingColumn);
+        CellEditEvent<String,String> ev = new CellEditEvent<String, String>(source, pos, editAnyEvent(), editedValue);
+        if (source != null) {
+            assertEquals(source, ev.getSource());
+        }
+        assertCellEditEventState(ev, table, editingColumn, pos, editedValue, oldValue, rowValue);
+    }
+
+    /**
+     * Asserts all properties of the event against the given expected values.
+     */
+    private <S, T> void assertCellEditEventState(CellEditEvent<S, T> event,
+            TreeTableView<S> table, TreeTableColumn<S, T> tableColumn, TreeTablePosition<S, T> pos,
+            T newValue, T oldValue, TreeItem<S> rowValue) {
+        assertEquals(newValue, event.getNewValue());
+        assertEquals(oldValue, event.getOldValue());
+        assertEquals(rowValue, event.getRowValue());
+        assertEquals(tableColumn, event.getTableColumn());
+        assertEquals(pos, event.getTreeTablePosition());
+        assertEquals(table, event.getTreeTableView());
+    }
+
+//------------ init
+
+    @Before public void setup() {
+        Thread.currentThread().setUncaughtExceptionHandler((thread, throwable) -> {
+            if (throwable instanceof RuntimeException) {
+                throw (RuntimeException)throwable;
+            } else {
+                Thread.currentThread().getThreadGroup().uncaughtException(thread, throwable);
+            }
+        });
+
+        TreeItem<String> root = new TreeItem<>("root");
+        root.setExpanded(true);
+        ObservableList<String> model = FXCollections.observableArrayList("Four", "Five", "Fear");
+            // "Flop", "Food", "Fizz"
```

Comment:
Did you forget to remove this comment?

Reply:
yeah, pitfalls of c&p test boiler-plate code ;)